File: android/src/main/kotlin/com/superwall/superwallkit_flutter/json/LogLevel+Json.kt
import com.superwall.sdk.logger.LogLevel
import com.superwall.superwallkit_flutter.json.JsonExtensions

fun JsonExtensions.Companion.logLevelFromJson(json: String): LogLevel? =
    when (json.lowercase()) {
        "debug" -> LogLevel.debug
        "info" -> LogLevel.info
        "warn" -> LogLevel.warn
        "error" -> LogLevel.error
        "none" -> LogLevel.none
        else -> null
    }

fun LogLevel.toJson(): String =
    when (this) {
        LogLevel.debug -> "debug"
        LogLevel.info -> "info"
        LogLevel.warn -> "warn"
        LogLevel.error -> "error"
        LogLevel.none -> "none"
        else -> "unknown"
    }

File: ios/Classes/Json/LogLevel+Json.swift
import SuperwallKit

extension LogLevel {
  static func fromJson(_ json: String) -> LogLevel? {
    switch json {
      case "debug":
        return LogLevel.debug
      case "info":
        return LogLevel.info
      case "warn":
        return LogLevel.warn
      case "error":
        return LogLevel.error
      case "none":
        return LogLevel.none
      default:
        return nil
    }
  }
}
